stan/math/prim/fun/mdivide_left_ldlt.hpp

```diff
@@ -13,29 +13,28 @@ namespace math {
 /**
  * Returns the solution of the system Ax=b given an LDLT_factor of A
  *
- * @tparam T type of elements in the LDLT_factor
- * @tparam R number of rows in the LDLT_factor, can be Eigen::Dynamic
- * @tparam C number of columns in the LDLT_factor, can be Eigen::Dynamic
- * @tparam EigMat type of the matrix
+ * @tparam T type of matrix in the LDLT_factor
+ * @tparam EigMat type of the right hand side
  *
  * @param A LDLT_factor
  * @param b Right hand side matrix or vector.
  * @return x = A^-1 b, solution of the linear system.
  * @throws std::domain_error if rows of b don't match the size of A.
  */
-template <int R, int C, typename T, typename EigMat,
-          require_eigen_t<EigMat>* = nullptr,
+template <typename T, typename EigMat, require_eigen_t<EigMat>* = nullptr,
           require_all_not_st_var<T, EigMat>* = nullptr,
-          require_any_not_t<std::is_arithmetic<T>,
+          require_any_not_t<std::is_arithmetic<value_type_t<T>>,
                             is_fvar<value_type_t<EigMat>>>* = nullptr>
-inline Eigen::Matrix<return_type_t<T, EigMat>, R, EigMat::ColsAtCompileTime>
-mdivide_left_ldlt(const LDLT_factor<T, R, C>& A, const EigMat& b) {
-  check_multiplicable("mdivide_left_ldlt", "A", A, "b", b);
-  if (A.cols() == 0) {
+inline Eigen::Matrix<return_type_t<T, EigMat>, Eigen::Dynamic,
+                     EigMat::ColsAtCompileTime>
+mdivide_left_ldlt(const LDLT_factor<T>& A, const EigMat& b) {
+  check_multiplicable("mdivide_left_ldlt", "A", A.matrix(), "b", b);
+
+  if (A.matrix().cols() == 0) {
     return {0, b.cols()};
   }
 
-  return A.solve(
+  return A.ldlt().solve(
       Eigen::Matrix<return_type_t<T, EigMat>, EigMat::RowsAtCompileTime,
                     EigMat::ColsAtCompileTime>(b));
```

Comment:
Good point. I'm gonna leave it as is though. This is some weird type promotion thing so the Eigen solves work. Like if `T` and `EigMat` don't match, it will make them match.

If we added another overload for when the scalar types are the same it would handle this. I grepped for where this function is used (it's an internal function not exposed to the Stan langage) and I don't think it's actually used anywhere (`mdivide_right_ldlt` uses this, but that is unused, and again that is an internal function), so I'm not gonna worry about this (though it would take about 5s to implement -- just seems like more code and complexity some future person has to decode).